subsys/bluetooth/mesh/beacon.c

```diff
@@ -744,6 +768,7 @@ BT_MESH_SUBNET_CB_DEFINE(beacon) = {
 void bt_mesh_beacon_init(void)
 {
 	k_work_init_delayable(&beacon_timer, beacon_send);
+	k_work_init_delayable(&beacon_send_work, beacon_send_work_cb);
```

Comment:
Can't this be handled in `beacon_send` with some 1 byte variable describing current action?